include/ThemeManager.hpp

```diff
@@ -0,0 +1,48 @@
+/*
+** Project FileShare, 2022
+**
+** Author Léo Lhuile
+**
+** Started on  Thu Jul 24 15:44:13 2025 Léo Lhuile
+** Last update Thu Jul 24 15:44:13 2025 Léo Lhuile
+**
+** ThemeManager.hpp : Theme management system
+*/
+
+#pragma once
+
+#include <string>
+#include <TGUI/TGUI.hpp>
+
+#ifdef _WIN32
+#include <windows.h>
+#include <winreg.h>
+#elif __APPLE__
+#include <CoreFoundation/CoreFoundation.h>
+#elif __linux__
+#include <cstdlib>
+#include <fstream>
+#include <algorithm>
+#endif
```

Comment:
I believe that indenting preprocessor #if makes them much more readable.
Also, instead of relying on the raw OS detection macros, we have `<CppSockets/OSDetection.hpp>` which defines `OS_LINUX`, `OS_WINDOWS`, and `OS_APPLE` which are easier to remember

```suggestion
<CppSockets/OSDetection.hpp>

#ifdef OS_WINDOWS
  #include <windows.h>
  #include <winreg.h>
#elif OS_APPLE
  #include <CoreFoundation/CoreFoundation.h>
#elif OS_LINUX
  #include <cstdlib>
  #include <fstream>
  #include <algorithm>
#endif
```